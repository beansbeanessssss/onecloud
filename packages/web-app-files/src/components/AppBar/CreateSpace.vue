<template>
  <oc-button
    id="new-space-menu-btn"
    key="new-space-menu-btn-enabled"
    v-oc-tooltip="$gettext('Create a new space')"
    :aria-label="$gettext('Create a new space')"
    appearance="raw"
    variation="inverse"
    class="oc-background-primary-gradient oc-px-s oc-text-nowrap oc-py-s"
    @click="showCreateSpaceModal"
  >
    <oc-icon name="add" />
    <translate>New Space</translate>
  </oc-button>
</template>

<script lang="ts">
import { mapActions, mapMutations } from 'vuex'

import { defineComponent } from '@vue/composition-api'
import { useGraphClient } from 'web-client/src/composables'
import { buildSpace } from 'web-client/src/helpers'
import { configurationManager } from 'web-pkg/src/configuration'

export default defineComponent({
  setup() {
    return {
      ...useGraphClient()
    }
  },
  methods: {
    ...mapActions(['showMessage', 'createModal', 'hideModal', 'setModalInputErrorMessage']),
    ...mapMutations('runtime/spaces', ['UPSERT_SPACE']),
    ...mapMutations('Files', [
      'SET_CURRENT_FOLDER',
      'LOAD_FILES',
      'CLEAR_CURRENT_FILES_LIST',
      'SET_FILE_SELECTION',
      'UPSERT_RESOURCE',
      'UPDATE_RESOURCE_FIELD'
    ]),

    showCreateSpaceModal() {
      const modal = {
        variation: 'passive',
        title: this.$gettext('Create a new space'),
        cancelText: this.$gettext('Cancel'),
        confirmText: this.$gettext('Create'),
        hasInput: true,
        inputLabel: this.$gettext('Space name'),
        inputValue: this.$gettext('New space'),
        onCancel: this.hideModal,
        onConfirm: this.addNewSpace,
        onInput: this.checkSpaceName
      }

      this.createModal(modal)
    },

    checkSpaceName(name) {
      if (name.trim() === '') {
        this.setModalInputErrorMessage(this.$gettext('Space name cannot be empty'))
        return
      }
      return this.setModalInputErrorMessage(null)
    },

    addNewSpace(name) {
      return this.graphClient.drives
        .createDrive({ name }, {})
        .then(({ data: space }) => {
          this.hideModal()
          const resource = buildSpace({ ...space, serverUrl: configurationManager.serverUrl })
          this.UPSERT_RESOURCE(resource)
          this.UPSERT_SPACE(resource)

          this.$client.files.createFolder(`spaces/${space.id}/.space`).then(() => {
            this.$client.files
              .putFileContents(
                `spaces/${space.id}/.space/readme.md`,
                `### ðŸ‘‹ ${this.$gettext('Hello!')}\r\n${this.$gettext(
                  'Add a description to welcome the members of the Space.'
                )}\r\n${this.$gettext(
                  'Use markdown to format your text. [More info]'
                )}(https://www.markdownguide.org/basic-syntax/)`
              )
              .then((markdown) => {
                this.graphClient.drives
                  .updateDrive(
                    space.id,
                    {
                      special: [
                        {
                          specialFolder: {
                            name: 'readme'
                          },
                          id: markdown['OC-FileId']
                        }
                      ]
                    },
                    {}
                  )
                  .then(({ data }) => {
                    this.UPDATE_RESOURCE_FIELD({
                      id: space.id,
                      field: 'spaceReadmeData',
                      value: data.special.find((special) => special.specialFolder.name === 'readme')
                    })
                  })
              })
          })
        })
        .catch((error) => {
          console.error(error)
          this.showMessage({
            title: this.$gettext('Creating space failedâ€¦'),
            status: 'danger'
          })
        })
    }
  }
})
</script>
