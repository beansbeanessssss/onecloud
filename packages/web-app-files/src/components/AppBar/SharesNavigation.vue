<template>
  <nav id="shares-navigation" class="oc-py-s" :aria-label="$gettext('Shares pages navigation')">
    <oc-list class="oc-flex oc-visible@s">
      <li v-for="navItem in navItems" :key="`shares-navigation-desktop-${navItem.to}`">
        <oc-button
          type="router-link"
          class="oc-mr-s"
          appearance="raw"
          :variation="navItem.active ? 'primary' : 'passive'"
          :to="navItem.to"
        >
          <span v-text="navItem.text" />
        </oc-button>
      </li>
    </oc-list>
    <div class="oc-hidden@s">
      <oc-button id="shares_navigation_mobile" appearance="raw" v-text="$gettext('Shares pages')" />
      <oc-drop toggle="#shares_navigation_mobile" mode="click" close-on-click padding-size="small">
        <oc-list>
          <li v-for="navItem in navItems" :key="`shares-navigation-mobile-${navItem.to}`">
            <oc-button
              type="router-link"
              appearance="raw"
              :variation="navItem.active ? 'primary' : 'passive'"
              :to="navItem.to"
            >
              <span v-text="navItem.text" />
            </oc-button>
          </li>
        </oc-list>
      </oc-drop>
    </div>
  </nav>
</template>

<script>
import {
  isLocationSharesActive,
  locationSharesViaLink,
  locationSharesWithMe,
  locationSharesWithOthers
} from '../../router/shares'
import { computed, getCurrentInstance } from '@vue/composition-api'
import { useRouter } from 'web-pkg/src/composables'

export default {
  setup() {
    const $gettext = getCurrentInstance().proxy.$gettext
    const router = useRouter()
    const sharesRoutes = [
      locationSharesWithMe,
      locationSharesWithOthers,
      locationSharesViaLink
    ].reduce((routes, route) => {
      routes[route.name] = router.getRoutes().find((r) => r.name === route.name)
      return routes
    }, {})
    const navItems = computed(() => [
      {
        to: sharesRoutes[locationSharesWithMe.name].path,
        text: $gettext('Shared with me'),
        active: isLocationSharesActive(router, 'files-shares-with-me')
      },
      {
        to: sharesRoutes[locationSharesWithOthers.name].path,
        text: $gettext('Shared with others'),
        active: isLocationSharesActive(router, 'files-shares-with-others')
      },
      {
        to: sharesRoutes[locationSharesViaLink.name].path,
        text: $gettext('Shared via link'),
        active: isLocationSharesActive(router, 'files-shares-via-link')
      }
    ])
    return {
      navItems
    }
  }
}
</script>
<style lang="scss" scoped>
.router-link-active {
  text-decoration: underline;
}
</style>
