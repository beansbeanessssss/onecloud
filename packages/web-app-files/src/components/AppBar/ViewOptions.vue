<template>
  <div class="oc-flex oc-flex-middle">
    <div
      v-if="viewModes.length"
      data-testid="viewmode-switch-buttons"
      class="oc-button-group oc-visible@s oc-mr-s"
    >
      <oc-button
        v-for="viewMode in viewModes"
        :key="viewMode.name"
        v-oc-tooltip="$gettext(viewMode.label)"
        :appearance="viewModeCurrent === viewMode.name ? 'filled' : 'outline'"
        :label="$gettext(viewMode.label)"
        @click="setViewMode(viewMode)"
      >
        <oc-icon :name="viewMode.icon.name" :fill-type="viewMode.icon.fillType" size="small" />
      </oc-button>
    </div>
    <oc-button
      id="files-view-options-btn"
      key="files-view-options-btn"
      v-oc-tooltip="viewOptionsButtonLabel"
      data-testid="files-view-options-btn"
      :aria-label="viewOptionsButtonLabel"
      appearance="raw"
      class="oc-my-s oc-p-xs"
    >
      <oc-icon name="settings-3" fill-type="line" />
    </oc-button>
    <oc-drop
      drop-id="files-view-options-drop"
      toggle="#files-view-options-btn"
      mode="click"
      class="oc-width-auto"
      padding-size="medium"
    >
      <oc-list>
        <li class="files-view-options-list-item oc-mb-m">
          <oc-switch
            v-model="hiddenFilesShownModel"
            data-testid="files-switch-hidden-files"
            :label="$gettext('Show hidden files')"
          />
        </li>
        <li class="files-view-options-list-item oc-my-m">
          <oc-switch
            v-model="fileExtensionsShownModel"
            data-testid="files-switch-files-extensions-files"
            :label="$gettext('Show file extensions')"
          />
        </li>
        <li class="files-view-options-list-item oc-mt-m">
          <oc-page-size
            :selected="itemsPerPage"
            data-testid="files-pagination-size"
            :label="$gettext('Items per page')"
            :options="[100, 500]"
            class="files-pagination-size"
            @change="itemsPerPage = $event"
          />
        </li>
      </oc-list>
    </oc-drop>
  </div>
</template>

<script lang="ts">
import { PropType } from 'vue'
import { mapMutations, mapState } from 'vuex'
import { useRouteQueryPersisted } from 'web-pkg/src/composables'
import { ViewMode } from 'web-pkg/src/ui/types'
import { PaginationConstants, ViewModeConstants } from '../../composables'
import { defineComponent } from 'vue'

export default defineComponent({
  props: {
    viewModes: {
      type: Array as PropType<ViewMode[]>,
      default: () => []
    }
  },
  setup() {
    const perPageQuery = useRouteQueryPersisted({
      name: PaginationConstants.perPageQueryName,
      defaultValue: PaginationConstants.perPageDefault
    })
    const viewModeQuery = useRouteQueryPersisted({
      name: ViewModeConstants.queryName,
      defaultValue: ViewModeConstants.defaultModeName
    })

    return {
      ViewModeConstants,
      viewModeCurrent: viewModeQuery,
      itemsPerPage: perPageQuery
    }
  },
  computed: {
    ...mapState('Files', ['areHiddenFilesShown', 'areFileExtensionsShown']),

    viewOptionsButtonLabel() {
      return this.$gettext('Display customization options of the files list')
    },

    hiddenFilesShownModel: {
      get() {
        return this.areHiddenFilesShown
      },

      set(value) {
        this.SET_HIDDEN_FILES_VISIBILITY(value)
      }
    },
    fileExtensionsShownModel: {
      get() {
        return this.areFileExtensionsShown
      },

      set(value) {
        this.SET_FILE_EXTENSIONS_VISIBILITY(value)
      }
    }
  },
  methods: {
    ...mapMutations('Files', ['SET_HIDDEN_FILES_VISIBILITY', 'SET_FILE_EXTENSIONS_VISIBILITY']),
    setViewMode(mode) {
      this.viewModeCurrent = mode.name
    }
  }
})
</script>

<style lang="scss" scoped>
#files-view-options-btn {
  vertical-align: middle;
  border: 3px solid transparent;
  &:hover {
    background-color: var(--oc-color-background-hover);
    border-radius: 3px;
  }
}
.files-view-options-list-item {
  & > * {
    display: flex;
    justify-content: space-between;
  }

  & + & {
    margin-top: var(--oc-space-small);
  }
}
</style>
