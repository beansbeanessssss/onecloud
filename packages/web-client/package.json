{
  "name": "@ownclouders/web-client",
  "version": "0.0.1",
  "description": "ownCloud web client",
  "license": "AGPL-3.0",
  "private": false,
  "author": "ownCloud GmbH <devops@owncloud.com>",
  "homepage": "https://github.com/owncloud/web/tree/master/packages/web-client",
  "repository": {
    "type": "git",
    "url": "https://github.com/owncloud/web",
    "directory": "packages/web-client"
  },
  "exports": {
    ".": {
      "default": "./src/index.ts"
    },
    "./graph": {
      "default": "./src/graph/index.ts"
    },
    "./graph/generated": {
      "default": "./src/graph/generated/index.ts"
    },
    "./ocs": {
      "default": "./src/ocs/index.ts"
    },
    "./sse": {
      "default": "./src/sse/index.ts"
    },
    "./webdav": {
      "default": "./src/webdav/index.ts"
    },
    "./l10n": {
      "default": "./l10n/translations.json"
    }
  },
  "scripts": {
    "generate-openapi": "rm -rf src/generated && docker run --rm -v \"${PWD}/src:/local\" openapitools/openapi-generator-cli generate -i https://raw.githubusercontent.com/owncloud/libre-graph-api/main/api/openapi-spec/v1.0.yaml  -g typescript-axios -o /local/generated"
  },
  "peerDependencies": {
    "@ownclouders/web-client": "workspace:*",
    "@casl/ability": "^6.3.3",
    "axios": "1.6.8",
    "fast-xml-parser": "4.3.3",
    "lodash-es": "^4.17.21",
    "luxon": "3.2.1",
    "@microsoft/fetch-event-source": "^2.0.1",
    "webdav": "5.3.1",
    "xml-js": "^1.6.11",
    "zod": "3.22.4"
  }
}
