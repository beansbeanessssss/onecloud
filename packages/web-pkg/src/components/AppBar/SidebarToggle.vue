<template>
  <oc-button
    id="files-toggle-sidebar"
    v-oc-tooltip="toggleSidebarButtonLabel"
    :aria-label="toggleSidebarButtonLabel"
    appearance="raw"
    class="oc-my-s oc-p-xs"
    @click.stop="toggleSideBar"
  >
    <oc-icon name="side-bar-right" :fill-type="toggleSidebarButtonIconFillType" />
  </oc-button>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { eventBus } from '../../services/eventBus'
import { SideBarEventTopics } from '../../composables/sideBar'

export default defineComponent({
  props: {
    isSideBarOpen: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    toggleSidebarButtonLabel() {
      if (this.isSideBarOpen) {
        return this.$gettext('Close sidebar to hide details')
      }
      return this.$gettext('Open sidebar to view details')
    },

    toggleSidebarButtonIconFillType() {
      return this.isSideBarOpen ? 'fill' : 'line'
    }
  },
  methods: {
    toggleSideBar() {
      eventBus.publish(SideBarEventTopics.toggle)
    }
  }
})
</script>

<style lang="scss" scoped>
#files-toggle-sidebar {
  vertical-align: middle;
  border: 3px solid transparent;
  &:hover {
    background-color: var(--oc-color-background-hover);
    border-radius: 3px;
  }
}
</style>
