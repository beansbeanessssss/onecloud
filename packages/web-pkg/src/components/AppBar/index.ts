export { default as AppBar } from './AppBar.vue'
export { default as SidebarToggle } from './SidebarToggle.vue'
