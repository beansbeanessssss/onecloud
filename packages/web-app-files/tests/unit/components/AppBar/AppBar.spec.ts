import AppBar from 'web-app-files/src/components/AppBar/AppBar.vue'
import { mockDeep } from 'jest-mock-extended'
import { Resource } from 'web-client'
import {
  createStore,
  defaultComponentMocks,
  defaultPlugins,
  shallowMount,
  defaultStoreMockOptions
} from 'web-test-helpers'

const selectedFiles = [mockDeep<Resource>(), mockDeep<Resource>()]
const actionSlot = "<button class='action-slot'>Click</button>"
const contextMenuSlot = "<button class='context-menu-slot'>Click</button>"
const contentSlot = "<div class='content-slot'>Foo</div>"

const breadcrumbItems = [
  { text: 'Example1', to: '/' },
  { text: 'Example2', to: '/foo' }
]
const breadCrumbItemWithContextActionAllowed = {
  text: 'Example Special',
  to: '/bar',
  allowContextActions: true
}

window.ResizeObserver =
  window.ResizeObserver ||
  jest.fn().mockImplementation(() => ({
    disconnect: jest.fn(),
    observe: jest.fn(),
    unobserve: jest.fn()
  }))

describe('AppBar component', () => {
  describe('renders', () => {
    it('by default no breadcrumbs, no bulkactions, no sharesnavigation but viewoptions and sidebartoggle', () => {
      const { wrapper } = getShallowWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
    describe('breadcrumbs', () => {
      it('if given, by default without breadcrumbsContextActionsItems', () => {
        const { wrapper } = getShallowWrapper([], {}, { breadcrumbs: breadcrumbItems })
        expect(wrapper.html()).toMatchSnapshot()
      })
      it('if given, with breadcrumbsContextActionsItems if allowed on last breadcrumb item', () => {
        const { wrapper } = getShallowWrapper(
          [],
          {},
          { breadcrumbs: [...breadcrumbItems, breadCrumbItemWithContextActionAllowed] }
        )
        expect(wrapper.html()).toMatchSnapshot()
      })
      it('if given, with content in the contextMenu slot', () => {
        const { wrapper } = getShallowWrapper(
          [],
          { contextMenu: contextMenuSlot },
          { breadcrumbs: [...breadcrumbItems, breadCrumbItemWithContextActionAllowed] }
        )
        expect(wrapper.html()).toMatchSnapshot()
      })
    })
    describe('bulkActions', () => {
      it('if enabled', () => {
        const { wrapper } = getShallowWrapper(selectedFiles, {}, { hasBulkActions: true })
        expect(wrapper.html()).toMatchSnapshot()
      })
      it('not if 1 file selected', () => {
        const { wrapper } = getShallowWrapper([selectedFiles[0]], {}, { hasBulkActions: true })
        expect(wrapper.html()).toMatchSnapshot()
      })
    })
    describe('sharesNavigation', () => {
      it('if enabled', () => {
        const { wrapper } = getShallowWrapper([], {}, { hasSharesNavigation: true })
        expect(wrapper.html()).toMatchSnapshot()
      })
    })
    describe('viewoptions and sidebartoggle', () => {
      it('only viewoptions if sidebartoggle is disabled', () => {
        const { wrapper } = getShallowWrapper([], {}, { hasSidebarToggle: false })
        expect(wrapper.html()).toMatchSnapshot()
      })
      it('only sidebartoggle if viewoptions is disabled', () => {
        const { wrapper } = getShallowWrapper([], {}, { hasViewOptions: false })
        expect(wrapper.html()).toMatchSnapshot()
      })
      it('neither if both are disabled', () => {
        const { wrapper } = getShallowWrapper(
          [],
          {},
          { hasSidebarToggle: false, hasViewOptions: false }
        )
        expect(wrapper.html()).toMatchSnapshot()
      })
      it('passes viewModes array to ViewOptions if displayViewModeSwitch props is enabled', () => {
        const { wrapper } = getShallowWrapper(
          [],
          {},
          { displayViewModeSwitch: true, hasViewOptions: true }
        )
        expect(wrapper).toMatchSnapshot()
      })
    })
    it('if given, with content in the actions slot', () => {
      const { wrapper } = getShallowWrapper([], { actions: actionSlot })
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('if given, with content in the content slot', () => {
      const { wrapper } = getShallowWrapper([], { content: contentSlot })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})

function getShallowWrapper(
  selected = [],
  slots = {},
  props: { [key: string]: any } = {
    breadcrumbs: [],
    displayViewModeSwitch: false,
    hasBulkActions: false,
    hasSharesNavigation: false,
    hasSidebarToggle: true,
    hasViewOptions: true
  }
) {
  const mocks = defaultComponentMocks({
    gettext: false,
    currentRoute: { name: 'files-trash-generic' }
  })
  mocks.$route.meta.title = 'ExampleTitle'
  const storeOptions = defaultStoreMockOptions
  storeOptions.modules.Files.getters.selectedFiles.mockImplementation(() => selected)
  const store = createStore(storeOptions)
  return {
    wrapper: shallowMount(AppBar, {
      props: { ...props },
      slots,
      global: {
        plugins: [...defaultPlugins(), store],
        mocks
      }
    })
  }
}
