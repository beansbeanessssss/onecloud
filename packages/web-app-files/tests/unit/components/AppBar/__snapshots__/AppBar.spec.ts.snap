// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar component renders breadcrumbs if given, by default without breadcrumbsContextActionsItems 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="No items selected." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-between">
      <oc-breadcrumb-stub class="oc-flex oc-flex-middle" contextmenupadding="small" id="files-breadcrumb" items="[object Object],[object Object]" variation="default"></oc-breadcrumb-stub>
      <!--v-if-->
      <div class="oc-flex">
        <view-options-stub viewmodes=""></view-options-stub>
        <sidebar-toggle-stub></sidebar-toggle-stub>
      </div>
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <!--v-if-->
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar component renders breadcrumbs if given, with breadcrumbsContextActionsItems if allowed on last breadcrumb item 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="No items selected." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-between">
      <oc-breadcrumb-stub class="oc-flex oc-flex-middle" contextmenupadding="small" id="files-breadcrumb" items="[object Object],[object Object],[object Object]" variation="default"></oc-breadcrumb-stub>
      <!--v-if-->
      <div class="oc-flex">
        <view-options-stub viewmodes=""></view-options-stub>
        <sidebar-toggle-stub></sidebar-toggle-stub>
      </div>
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <!--v-if-->
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar component renders breadcrumbs if given, with content in the contextMenu slot 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="No items selected." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-between">
      <oc-breadcrumb-stub class="oc-flex oc-flex-middle" contextmenupadding="small" id="files-breadcrumb" items="[object Object],[object Object],[object Object]" variation="default"></oc-breadcrumb-stub>
      <!--v-if-->
      <div class="oc-flex">
        <view-options-stub viewmodes=""></view-options-stub>
        <sidebar-toggle-stub></sidebar-toggle-stub>
      </div>
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <!--v-if-->
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar component renders bulkActions if enabled 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="2 items selected. Actions are available above the table." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-right">
      <!--v-if-->
      <!--v-if-->
      <div class="oc-flex">
        <view-options-stub viewmodes=""></view-options-stub>
        <sidebar-toggle-stub></sidebar-toggle-stub>
      </div>
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <!--v-if-->
        <batch-actions-stub></batch-actions-stub>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar component renders bulkActions not if 1 file selected 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="1 item selected. Actions are available above the table." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-right">
      <!--v-if-->
      <!--v-if-->
      <div class="oc-flex">
        <view-options-stub viewmodes=""></view-options-stub>
        <sidebar-toggle-stub></sidebar-toggle-stub>
      </div>
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <!--v-if-->
        <batch-actions-stub></batch-actions-stub>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar component renders by default no breadcrumbs, no bulkactions, no sharesnavigation but viewoptions and sidebartoggle 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="No items selected." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-right">
      <!--v-if-->
      <!--v-if-->
      <div class="oc-flex">
        <view-options-stub viewmodes=""></view-options-stub>
        <sidebar-toggle-stub></sidebar-toggle-stub>
      </div>
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <!--v-if-->
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar component renders if given, with content in the actions slot 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="No items selected." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-right">
      <!--v-if-->
      <!--v-if-->
      <div class="oc-flex">
        <view-options-stub viewmodes=""></view-options-stub>
        <sidebar-toggle-stub></sidebar-toggle-stub>
      </div>
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <button class="action-slot">Click</button>
        <!--v-if-->
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar component renders if given, with content in the content slot 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="No items selected." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-right">
      <!--v-if-->
      <!--v-if-->
      <div class="oc-flex">
        <view-options-stub viewmodes=""></view-options-stub>
        <sidebar-toggle-stub></sidebar-toggle-stub>
      </div>
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <!--v-if-->
      </div>
    </div>
    <div class="content-slot">Foo</div>
  </div>
</div>
`;

exports[`AppBar component renders sharesNavigation if enabled 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="No items selected." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-between">
      <!--v-if-->
      <shares-navigation-stub></shares-navigation-stub>
      <div class="oc-flex">
        <view-options-stub viewmodes=""></view-options-stub>
        <sidebar-toggle-stub></sidebar-toggle-stub>
      </div>
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <!--v-if-->
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar component renders viewoptions and sidebartoggle neither if both are disabled 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="No items selected." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-right">
      <!--v-if-->
      <!--v-if-->
      <!--v-if-->
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <!--v-if-->
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar component renders viewoptions and sidebartoggle only sidebartoggle if viewoptions is disabled 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="No items selected." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-right">
      <!--v-if-->
      <!--v-if-->
      <div class="oc-flex">
        <!--v-if-->
        <sidebar-toggle-stub></sidebar-toggle-stub>
      </div>
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <!--v-if-->
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar component renders viewoptions and sidebartoggle only viewoptions if sidebartoggle is disabled 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="No items selected." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-right">
      <!--v-if-->
      <!--v-if-->
      <div class="oc-flex">
        <view-options-stub viewmodes=""></view-options-stub>
        <!--v-if-->
      </div>
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <!--v-if-->
      </div>
    </div>
  </div>
</div>
`;

exports[`AppBar component renders viewoptions and sidebartoggle passes viewModes array to ViewOptions if displayViewModeSwitch props is enabled 1`] = `
<div class="" id="files-app-bar">
  <oc-hidden-announcer-stub announcement="No items selected." level="polite"></oc-hidden-announcer-stub>
  <div class="files-topbar oc-py-s">
    <h1 class="oc-invisible-sr">ExampleTitle</h1>
    <div class="oc-flex oc-flex-right">
      <!--v-if-->
      <!--v-if-->
      <div class="oc-flex">
        <view-options-stub viewmodes="[object Object],[object Object]"></view-options-stub>
        <sidebar-toggle-stub></sidebar-toggle-stub>
      </div>
    </div>
    <div class="files-app-bar-actions oc-mt-xs">
      <div class="oc-flex-1 oc-flex oc-flex-start">
        <!--v-if-->
      </div>
    </div>
  </div>
</div>
`;
